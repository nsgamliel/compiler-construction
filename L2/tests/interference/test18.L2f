  (:go
0

    // Save the callee save registers
%variableK2JK7XK90G <- r12

    // Allocate the data
    rdi <- 5
    rsi <- 5
    call allocate 2
    r12 <- rax

    // Print the data
    rdi <- r12
    mem rsp -8 <- :printData_ret
    call :printData 1
    :printData_ret

    // Restore the callee save registers
r12 <- %variableK2JK7XK90G

    return
  )
