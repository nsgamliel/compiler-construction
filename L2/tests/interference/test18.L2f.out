r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %variableK2JK7XK90G
r11 r10 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %variableK2JK7XK90G
r12 r10 r11 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %variableK2JK7XK90G
r13 r10 r11 r12 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %variableK2JK7XK90G
r14 r10 r11 r12 r13 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %variableK2JK7XK90G
r15 r10 r11 r12 r13 r14 r8 r9 rax rbp rbx rcx rdi rdx rsi %variableK2JK7XK90G
r8 r10 r11 r12 r13 r14 r15 r9 rax rbp rbx rcx rdi rdx rsi %variableK2JK7XK90G
r9 r10 r11 r12 r13 r14 r15 r8 rax rbp rbx rcx rdi rdx rsi %variableK2JK7XK90G
rax r10 r11 r12 r13 r14 r15 r8 r9 rbp rbx rcx rdi rdx rsi %variableK2JK7XK90G
rbp r10 r11 r12 r13 r14 r15 r8 r9 rax rbx rcx rdi rdx rsi %variableK2JK7XK90G
rbx r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rcx rdi rdx rsi %variableK2JK7XK90G
rcx r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rdi rdx rsi %variableK2JK7XK90G
rdi r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdx rsi %variableK2JK7XK90G
rdx r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rsi %variableK2JK7XK90G
rsi r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx %variableK2JK7XK90G
%variableK2JK7XK90G r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
